tiledb/sm/array_schema/dimension.cc

```diff
@@ -1588,21 +1588,45 @@ Status Dimension::set_null_tile_extent_to_range() {
     return LOG_STATUS(Status::DimensionError(
         "Cannot set tile extent to domain range; Domain not set"));
 
+  typedef typename std::make_unsigned<T>::type unsigned_t;
+
   // Calculate new tile extent equal to domain range
   auto domain = (const T*)domain_.data();
-  T tile_extent = domain[1] - domain[0];
+  unsigned_t tile_extent = (unsigned_t)domain[1] - (unsigned_t)domain[0];
 
   // We need to add 1 for integral domains
-  if (std::is_integral<T>::value) {
-    // Check overflow before adding 1
-    if (tile_extent == std::numeric_limits<T>::max())
-      return LOG_STATUS(Status::DimensionError(
-          "Cannot set null tile extent to domain range; "
-          "Domain range exceeds domain type max numeric limit"));
-    ++tile_extent;
-    // After this, tile_extent = domain[1] - domain[0] + 1, which is the correct
-    // domain range
-  }
+
+  // Check overflow before adding 1
+  if (tile_extent == std::numeric_limits<unsigned_t>::max())
+    return LOG_STATUS(Status::DimensionError(
+        "Cannot set null tile extent to domain range; "
+        "Domain range exceeds domain type max numeric limit"));
+  ++tile_extent;
+  // After this, tile_extent = domain[1] - domain[0] + 1, which is the correct
+  // domain range
+
+  // Allocate space
+  uint64_t type_size = sizeof(T);
+  tile_extent_.resize(type_size);
+  std::memcpy(&tile_extent_[0], &tile_extent, type_size);
+
+  return Status::Ok();
+}
+
+template <class T, typename std::enable_if<!std::is_integral<T>::value>::type*>
+Status Dimension::set_null_tile_extent_to_range() {
+  // Applicable only to null extents
+  if (!tile_extent_.empty())
+    return Status::Ok();
+
+  // Check empty domain
+  if (domain_.empty())
+    return LOG_STATUS(Status::DimensionError(
+        "Cannot set tile extent to domain range; Domain not set"));
+
+  // Calculate new tile extent equal to domain range
+  auto domain = (const T*)domain_.data();
+  T tile_extent = domain[1] - domain[0];
```

Comment:
This can overflow and assign NaN to `tile_extent`. Better to have an error here than to track down the effects of bogus data everywhere else.

Reply:
This part I'm not touching anymore... We can fix some in another change

Reply:
OK.